src/dsp_tools/analyse_xml_data/extract_links_from_XML.py

```diff
@@ -6,62 +6,67 @@
 from lxml import etree
 from viztracer import VizTracer
 
-from dsp_tools.analyse_xml_data.models import ResptrLink, UploadResource, XMLLink
+from dsp_tools.analyse_xml_data.models import ResptrLink, XMLLink
 
 
-def _create_info_from_xml_for_graph(root: etree._Element) -> tuple[list[ResptrLink], list[XMLLink], set[str]]:
+def _create_info_from_xml_for_graph(
+    root: etree._Element,
+) -> tuple[etree._Element, list[ResptrLink], list[XMLLink], list[str]]:
     """Create instances of the classes ResptrLink and XMLLink from the root of the XML file."""
     resptr_instances = []
     xml_instances = []
-    all_resource_ids = set()
+    all_resource_ids = []
     for resource in root.iter(tag="{https://dasch.swiss/schema}resource"):
         resptr, xml, subject_id = _create_info_from_xml_for_graph_from_one_resource(resource)
-        if resptr:
-            resptr_instances.extend(resptr)
-        if xml:
-            xml_instances.extend(xml)
-        if subject_id:
-            all_resource_ids.add(subject_id)
-    return resptr_instances, xml_instances, set(all_resource_ids)
+        all_resource_ids.append(subject_id)
+        resptr_instances.extend(resptr)
+        xml_instances.extend(xml)
+    return root, resptr_instances, xml_instances, all_resource_ids
```

Comment:
why do you return the root here? In my logic, you should never have to return the thing you passed in

Reply:
That seems weird... if you do in-place modification, it should also apply to the caller side. Sorry I insist on that... but I take some offence by this pattern, because from the method signature it suggests that the passed object will not be modified (but a modified copy gets returned - by convention that's the reason to return something like that), so to me this is misleading behaviour.

Reply:
Weirdest thing. I tested it again with root and here it works. If I check for the UUID attribute in a element for example in test_create_info_from_xml_for_graph_from_one_resource. Then the UUID attribute is not there. But in root (I did not check separately) it works. Do you understand why?

Reply:
I have no idea... but if I had to guess, maybe if you are on the iterator, modifications do not get applied there right away?